# Data Collection Notebook
**LO7: 7.1** - Data Collection

## Objectives
- Collect and load telecom churn dataset from Kaggle
- Perform initial data inspection and quality assessment
- Clean and prepare data for EDA and ML pipelines

## Inputs
- WA_Fn-UseC_-Telco-Customer-Churn.csv from Kaggle

## Outputs
- Cleaned dataset
- Data quality report
- Initial insights for business requirements


In [7]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

import sys
sys.path.append('..')
from src.data_loader import load_data


## 1. Load Dataset


In [8]:
# Load data
df = pd.read_csv('../data/WA_Fn-UseC_-Telco-Customer-Churn.csv')
print(f"Dataset shape: {df.shape}")
df.head()


Dataset shape: (7043, 21)


customerID  gender  SeniorCitizen Partner Dependents  tenure PhoneService  \
0  7590-VHVEG  Female              0     Yes         No       1           No   
1  5575-GNVDE    Male              0      No         No      34          Yes   
2  3668-QPYBK    Male              0      No         No       2          Yes   
3  7795-CFOCW    Male              0      No         No      45           No   
4  9237-HQITU  Female              0      No         No       2          Yes   

      MultipleLines InternetService OnlineSecurity  ... DeviceProtection  \
0  No phone service             DSL             No  ...               No   
1                No             DSL            Yes  ...              Yes   
2                No             DSL            Yes  ...               No   
3  No phone service             DSL            Yes  ...              Yes   
4                No     Fiber optic             No  ...               No   

  TechSupport StreamingTV StreamingMovies        Contract PaperlessBilling  \
0          No          No              No  Month-to-month              Yes   
1          No          No              No        One year               No   
2          No          No              No  Month-to-month              Yes   
3         Yes          No              No        One year               No   
4          No          No              No  Month-to-month              Yes   

               PaymentMethod MonthlyCharges  TotalCharges Churn  
0           Electronic check          29.85         29.85    No  
1               Mailed check          56.95        1889.5    No  
2               Mailed check          53.85        108.15   Yes  
3  Bank transfer (automatic)          42.30       1840.75    No  
4           Electronic check          70.70        151.65   Yes  

[5 rows x 21 columns]

## 2. Data Inspection


In [9]:
# Dataset info
df.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7043 entries, 0 to 7042
Data columns (total 21 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   customerID        7043 non-null   object 
 1   gender            7043 non-null   object 
 2   SeniorCitizen     7043 non-null   int64  
 3   Partner           7043 non-null   object 
 4   Dependents        7043 non-null   object 
 5   tenure            7043 non-null   int64  
 6   PhoneService      7043 non-null   object 
 7   MultipleLines     7043 non-null   object 
 8   InternetService   7043 non-null   object 
 9   OnlineSecurity    7043 non-null   object 
 10  OnlineBackup      7043 non-null   object 
 11  DeviceProtection  7043 non-null   object 
 12  TechSupport       7043 non-null   object 
 13  StreamingTV       7043 non-null   object 
 14  StreamingMovies   7043 non-null   object 
 15  Contract          7043 non-null   object 
 16  PaperlessBilling  7043 non-null   object 


In [10]:
# Check missing values
missing = df.isnull().sum()
print("Missing values:")
print(missing[missing > 0])


Missing values:
Series([], dtype: int64)


## 3. Data Cleaning (LO7: 7.2)


In [11]:
# Convert TotalCharges to numeric
df['TotalCharges'] = pd.to_numeric(df['TotalCharges'], errors='coerce')

# Check for null values created
print(f"Null values in TotalCharges: {df['TotalCharges'].isnull().sum()}")

# Drop rows with missing TotalCharges
df = df.dropna(subset=['TotalCharges'])
print(f"Dataset shape after cleaning: {df.shape}")


Null values in TotalCharges: 11
Dataset shape after cleaning: (7032, 21)


In [12]:
# Calculate LTV for analysis
df['LTV'] = df['MonthlyCharges'] * df['tenure']
print(f"Mean LTV: ${df['LTV'].mean():.2f}")


Mean LTV: $2283.15


## Conclusions
- Dataset contains 7,043 customer records (after cleaning)
- 21 features including demographics, services, and billing information
- Target variable: Churn (Yes/No)
- Data is ready for exploratory analysis and modeling
